internal/generator/vector/helpers/helpers.go

```diff
@@ -56,3 +59,49 @@ func ListenOnAllLocalInterfacesAddress() string {
 func EscapeDollarSigns(s string) string {
 	return strings.ReplaceAll(s, "$", "$$")
 }
+
+// TransformUserTemplateToVRL converts the user entered template to VRL compatible syntax
+// Example: foo-{.log_type||"none"} -> "foo-" + to_string!(.log_type||"none")
```

Comment:
It may be that we need to reject these changes for 5.9 and earlier.  My brief review of the API is we don't support templating like 6.x and I'm hesitant to add that functionality here.